openpype/pipeline/workfile/path_resolving.py

```diff
@@ -534,3 +541,50 @@ def create_workdir_extra_folders(
         fullpath = os.path.join(workdir, subfolder)
         if not os.path.exists(fullpath):
             os.makedirs(fullpath)
+
+
+def get_last_workfile_representation(
+    project_name: str,
+    asset_name: str,
+    task_name: str,
+    asset_id=None,
+    asset_doc: dict = None,
+    fields: Iterable[str] = None,
+) -> dict:
+    """Get last published workfile representation.
+
+    Args:
+        project_name(str): Project name.
+        asset_name(str): Asset/Shot name.
+        task_name (str): Task name.
+        asset_id (Optional[Union[str, ObjectId]]): Asset id.
+            subset name. Defaults to None.
+        asset_doc (Optional[dict]): Asset doc. Defaults to None.
+        fields (Optional[Iterable[str]]): Fields that should be returned.
+            Defaults to None.
+
+    Returns:
+        dict: Last workfile representation.
+    """
+    if not asset_doc:
+        asset_doc = get_asset_by_name(project_name, asset_name)
+
+    last_version = get_last_version_by_subset_name(
+        project_name,
+        get_subset_name(
+            "workfile",
+            "",
```

Comment:
```suggestion
            "main",
```
I believe the default template doesn't use the variant in the workfile subset naming but I believe it _can_ be - since those usually default to "main" as far as I know we might just be 5% safer here if we keep that as the default for `variant`.

It would be much better if somehow we would be able to retrieve the subset correctly without having to rely on the explicit subset naming but somehow be able to find it based on accurate context however - but I suppose that could be something for later.